pkgs/tools/wayland/gtklock/default.nix

```diff
@@ -0,0 +1,48 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, pkgs
+, pam
+, scdoc
+, gtk3
+, pkg-config
+, gtk-layer-shell
+, wayland
+}:
+stdenv.mkDerivation rec {
+  pname = "gtklock";
+  version = "1.3.4";
+
+  src = fetchFromGitHub {
+    owner = "jovanlanik";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0m5z863ihaxy4460c06m3g1y01ivgj4hadq302pcfdmh2ipqcf1v";
+  };
+
+  buildInputs = [
+    scdoc
+  ];
+  nativeBuildInputs = [
+    wayland
+    gtk3
+    pkg-config
+    pam
+    gtk-layer-shell
+  ];
+
+  installFlags = [
+    "DESTDIR=$(out)"
+    "PREFIX="
+  ];
+
+  meta = with lib; {
+    description = "GTK-based lockscreen for Wayland";
+    longDescription = ''
+      Important note: for gtklock to work you need to set "security.pam.services.gtklock = {};" manually.
+    ''; # Following  nixpkgs/pkgs/applications/window-managers/sway/lock.nix
```

Comment:
Why not do a nixos module for this?

Reply:
Ummm, seems more like a home-manager thing to me?

Reply:
`security.pam.services` is a NixOS option, not a home-manager option, since it's a system level configuration.

Reply:
Yes and for system services we should provide our own things Independent of other projects.